gen/builders/state_tracker.go

```diff
@@ -23,37 +30,154 @@ import (
 // messages.
 type StateTracker struct {
 	vector *schema.TestVector
+	bc     *BuilderCommon
+
+	StateTreeVersion types.StateTreeVersion
+	ActorsVersion    actors.Version
 
 	Stores    *Stores
 	StateTree *state.StateTree
 	Driver    *conformance.Driver
 
 	CurrRoot cid.Cid
+
+	EmptyObjectCid       cid.Cid
+	EmptyArrayCid        cid.Cid
+	EmptyMapCid          cid.Cid
+	EmptyMultiMapCid     cid.Cid
+	EmptyBitfieldCid     cid.Cid
+	EmptyDeadlinesCid    cid.Cid
+	EmptyVestingFundsCid cid.Cid
 }
 
-func NewStateTracker(selector schema.Selector, vector *schema.TestVector) *StateTracker {
+func NewStateTracker(bc *BuilderCommon, selector schema.Selector, vector *schema.TestVector, stVersion types.StateTreeVersion, actorsVersion actors.Version, zeroFn ActorsZeroStateFn) *StateTracker {
 	stores := NewLocalStores(context.Background())
 
 	// create a brand new state tree.
-	// TODO: specify network version in vectors.
-	st, err := state.NewStateTree(stores.CBORStore, types.StateTreeVersion0)
+	st, err := state.NewStateTree(stores.CBORStore, stVersion)
 	if err != nil {
 		panic(err)
 	}
 
 	stkr := &StateTracker{
-		vector:    vector,
-		Stores:    stores,
-		StateTree: st,
-		Driver:    conformance.NewDriver(context.Background(), selector, conformance.DriverOpts{}),
+		vector:           vector,
+		bc:               bc,
+		StateTreeVersion: stVersion,
+		ActorsVersion:    actorsVersion,
+		Stores:           stores,
+		StateTree:        st,
+		Driver:           conformance.NewDriver(context.Background(), selector, conformance.DriverOpts{}),
 	}
 
-	stkr.initializeZeroState(selector)
+	stkr.initEmptyStructures()
+
+	zeroFn(stkr, selector)
 
 	_ = stkr.Flush()
 	return stkr
 }
 
+func (st *StateTracker) initEmptyStructures() {
+	// empty object
+	st.EmptyObjectCid = func() cid.Cid {
+		empty, err := st.Stores.ADTStore.Put(context.TODO(), []struct{}{})
+		if err != nil {
+			panic(err)
+		}
+		return empty
+	}()
+
+	// bitfield -- same no matter actors version.
+	st.EmptyBitfieldCid = func() cid.Cid {
+		empty, err := st.Stores.ADTStore.Put(context.TODO(), bitfield.NewFromSet(nil))
+		if err != nil {
+			panic(err)
+		}
+		return empty
+	}()
+
+	// map -- lotus abstraction for version selection EXISTS.
+	st.EmptyMapCid = func() cid.Cid {
+		empty, err := adt.NewMap(st.Stores.ADTStore, st.ActorsVersion)
+		if err != nil {
+			panic(err)
+		}
+		ret, err := empty.Root()
+		if err != nil {
+			panic(err)
+		}
+		return ret
+	}()
+
+	// array -- lotus abstraction for version selection EXISTS.
+	st.EmptyArrayCid = func() cid.Cid {
+		empty, err := adt.NewArray(st.Stores.ADTStore, st.ActorsVersion)
+		if err != nil {
+			panic(err)
+		}
+		ret, err := empty.Root()
+		if err != nil {
+			panic(err)
+		}
+		return ret
+	}()
+
+	// multimap -- lotus abstraction for version selection does NOT exist.
+	st.EmptyMultiMapCid = func() (ret cid.Cid) {
```

Comment:
Unfortunately, a Lotus abstraction does not exist for this, so we need to rely on specific actors versions.